Checkout/Checkout/Source/Validation/Validators/CardValidator.swift

```diff
@@ -178,63 +203,69 @@ public class CardValidator: CardValidating {
   // MARK: - Private
 
   private func validateExpiry(
-    expiryMonth: String,
-    expiryYear: String
+    expiryMonth: Int,
+    expiryYear: Int
   ) -> Result<ExpiryDate, ValidationError.ExpiryDate> {
-    guard let month = Int(expiryMonth) else {
-      return .failure(.invalidMonthString)
-    }
-
-    guard let year = Int(expiryYear) else {
-      return .failure(.invalidYearString)
-    }
-
-    guard month >= 1 && month <= 12 else {
+    guard expiryMonth >= 1 && expiryMonth <= 12 else {
       return .failure(.invalidMonth)
     }
 
-    let fourDigitYear: Int
-
-    switch self.fourDigitYear(
-      from: expiryYear,
-      year: year) {
-    case .success(let fourDigitYearResult):
-      fourDigitYear = fourDigitYearResult
-    case .failure(let expiryDateError):
-      return .failure(expiryDateError)
-    }
-
     let currentDate = Date()
+    // using UTC-12 as this is the latest timezone where a card could be valid
+    let cardExpiryComponents = DateComponents(
+      timeZone: .utcMinus12,
+      year: expiryYear,
+      month: expiryMonth
+    )
 
-    guard let providedDate = calendar.date(
-      from: DateComponents(
-        year: fourDigitYear,
-        month: month
-      )
-    ), providedDate >= currentDate else {
+    guard
+      let cardExpiryDate = calendar.date(from: cardExpiryComponents),
+      let nextMonth = calendar.date(byAdding: .month, value: 1, to: cardExpiryDate),
+      nextMonth > currentDate
+    else {
       return .failure(.inThePast)
     }
 
     let expiryDate = ExpiryDate(
-      month: month,
-      year: fourDigitYear)
+      month: expiryMonth,
+      year: expiryYear)
 
     return .success(expiryDate)
   }
 
-  private func fourDigitYear(
-    from expiryYear: String,
-    year: Int
-  ) -> Result<Int, ValidationError.ExpiryDate> {
+  private func fourDigitYear(from expiryYear: String) -> Result<Int, ValidationError.ExpiryDate> {
     switch expiryYear.count {
-    case 0, 1, 3:
-      return .failure(.incompleteYear)
-    case 2:
-      return .success(2000 + year)
-    case 4:
-      return .success(year)
+    case 0...4:
+      guard let year = Int(expiryYear) else {
+        return .failure(.invalidYearString)
+      }
+
+      return fourDigitYear(from: year)
     default:
       return .failure(.invalidYear)
     }
   }
+
+  private func fourDigitYear(from expiryYear: Int) -> Result<Int, ValidationError.ExpiryDate> {
```

Comment:
Just a side note that it feels weird to support 2 digit years. If you think while typing:
"2" = Invalid
"20" = "2020"
"203" = Invalid
"2034" = "2034"

If we're ok with this possible behaviour on the UX, no problem, but may just be worth being aware of possible uses of an expectation that 2 digits is safe.

Reply:
2 or 4 digits are both valid for expiry year. This is expected behaviour. Do you think in any case this will have issues ?

Reply:
If we have it all under control and implemented, no problem. But if we are going to update the UI to read this result on every input change and we are using it incorrectly it could lead to the chain I mentioned in previous message.